ui/page/root/home_page.go

```diff
@@ -810,3 +821,33 @@ func (hp *HomePage) CalculateAssetsUSDBalance() {
 		hp.ParentWindow().Reload()
 	}
 }
+
+// initializeDEX will initialize hp.dexCore if it has not been initialized
+// before. It's a no-op if hp.dexCore has been initialized.
+func (hp *HomePage) ensureDEXIsReady() {
+	if hp.dexCore != nil && hp.dexCoreRunning {
+		return // nothing to do
+	}
+
+	logger, logCloser, err := dexcore.NewDexLogger(filepath.Dir(hp.WL.AssetsManager.LogFile()), hp.WL.AssetsManager.GetLogLevels(), hp.WL.AssetsManager.MaxLogZips)
+	if err != nil {
+		log.Errorf("error initializing dex log: %v", err)
+		return
+	}
+
+	hp.dexCore, err = dexcore.New(hp.WL.AssetsManager.NetworkRootDir(), hp.WL.AssetsManager.GetLanguagePreference(), hp.WL.AssetsManager.NetType(), logger) // TODO: We should be able to update dex core's language when the user changes language.
+	if err != nil {
+		log.Errorf("error initializing dex core: %v", err)
+		return
+	}
+
+	hp.dexCoreRunning = true
+	// Use a goroutine to start dex core as it'll blocks until dex core exits.
+	go func() {
+		// TODO: We should be able to shutdown core properly if Homepage is ever
+		// destroyed and recreated while the app is still running.
+		hp.dexCore.Run(hp.dexCtx)
+		hp.dexCoreRunning = false
```

Comment:
Use a page ctx as mentioned in a previous comment.
Use a `wg` as well, increment before starting the goroutine, decrement inside the goroutine after dexCore.Run(.) returns.
In OnNavigatedFrom(), cancel the page ctx, and wait on the `wg`.

Reply:
It actually looks like all these can go to the dexcore package. A `StartDEXClient()` function that does all this and returns the DEXClient. Which reminds me, DEXCore should probably be DEXClient. `Core` is a dcrdex concept. In cryptopower, we're dealing with a DEXClient which uses dcrdex.Core under the hood.